doc/releases/index.rst

```diff
@@ -74,6 +74,12 @@ so with the help of west.
 The project's technical documentation is also tagged to correspond with a
 specific release and can be found at https://docs.zephyrproject.org/.
 
+Release notes contain a list of changes that have been made to the different
+areas of the project during the development cycle of the release. Those do not
+include changes that require the user to modify their own application in order
+to upgrade to the release. Those instead belong in the corresponding migration
+document.
```

Comment:
I was hoping for more recommendations on what to put where, whether to link between the two files, etc.

At least a link and an introduction to the split of documentation at the very top of each release note file would be good. I am afraid users/developers will overlook the newly introduced migration guide otherwise.

Reply:
Sorry that I, as the assignee, let this one go stale. I agree with @henrikbrixandersen.
Along these lines, I think the current wording implies that any change requiring modifications/migration would *only* be described in the migration guide, which I think would be counterproductive and don't reflect how we'd want to do things.
See e.g commit ac520e6f99ce477f512a797f90ec08b063850fe1: it does have a short entry in the release notes describing something new *and* a related entry in the migration guide as this new features comes with some breaking changes.

Something like this?

```suggestion
Release notes contain a list of changes that have been made to the different
areas of the project during the development cycle of the release. Changes that
require the user to modify their own application to support the new release are 
not to be detailed in the release notes but rather have dedicated entries in the
corresponding migration document
```